test/cpp_extensions/cpp_c10d_extension.cpp

```diff
@@ -0,0 +1,101 @@
+#include "cpp_c10d_extension.hpp"
+
+#include <map>
+
+namespace c10d {
+
+std::shared_ptr<ProcessGroup> ProcessGroupC10dTest::createProcessGroupC10dTest(
+    const std::shared_ptr<::c10d::Store>& store,
+    int rank,
+    int size,
+    const std::string& groupName = "") {
+  return std::make_shared<ProcessGroupC10dTest>(rank, size);
+}
+
+ProcessGroupC10dTest::ProcessGroupC10dTest(int rank, int size)
+    : ProcessGroup(rank, size) {}
+
+ProcessGroupC10dTest::~ProcessGroupC10dTest() {}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::broadcast(
+    std::vector<at::Tensor>& tensors,
+    const BroadcastOptions& opts) {
+  return NULL;
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::allreduce(
+    std::vector<at::Tensor>& tensors,
+    const AllreduceOptions& opts) {
+  return NULL;
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::reduce(
+    std::vector<at::Tensor>& tensors,
+    const ReduceOptions& opts) {
+  return NULL;
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::allgather(
+    std::vector<std::vector<at::Tensor>>& outputTensors,
+    std::vector<at::Tensor>& inputTensors,
+    const AllgatherOptions& opts) {
+  throw std::runtime_error("ProcessGroupC10dTest does not support allgather");
```

Comment:
Wonder why we throw exception in case of allgather but "return NULL" in case of allreduce?

Reply:
I updated the logic to return a dummy work for only broadcast() and allreduce(). others are still unsupported